flowey/flowey_core/src/util.rs

```diff
@@ -0,0 +1,18 @@
+//! Utilities used by flowey_core and made avaiable for higher-level crates.
+
+use std::path::Path;
+
+/// Copies the contents of `src` into the directory `dst`.
+pub fn copy_dir_all(src: impl AsRef<Path>, dst: impl AsRef<Path>) -> std::io::Result<()> {
```

Comment:
If you're going to do this, my suggestion is to have these functions accept a `_rt: &mut RustRuntimeServices<'_>` parameter, to act as a "proof" that the function is only called at runtime.

The long-term alternative would be to invest in some more interesting sandboxing during pipeline resolution time (e.g: seccomp filters to fail pipeline compilation if the code attempts to read/write the fs at compile time), but in the meantime, having some guard rails via type-system proof would be good.

Reply:
I dunno, man, we already can't stop people from reaching for `fs_err::copy` for single files.

I think the fact that you just don't _have_ the paths to copy to/from is probably enough, in practice.

Reply:
I was approaching it more from the lens of API consistency (wrt. some of the other helper APIs that surround it), but sure, these particular filesystem manipulation APIs are probably not going to be misused